hermes-benchmark/src/jmh/java/pl/allegro/tech/hermes/benchmark/environment/FrontendEnvironment.java

```diff
@@ -0,0 +1,100 @@
+import pl.allegro.tech.hermes.test.helper.environment.ZookeeperStarter;
+
+import java.io.IOException;
+
+import static pl.allegro.tech.hermes.api.ContentType.AVRO;
+import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic;
+
+@State(Scope.Benchmark)
+public class FrontendEnvironment {
+
+    private static final int MAX_CONNECTIONS_PER_ROUTE = 200;
+    private static final int ZOOKEEPER_PORT = 2181;
+
+    private HermesFrontend hermesFrontend;
+    private ZookeeperStarter zookeeperStarter;
+    private KafkaStarter kafkaStarter;
+    private HermesPublisher publisher;
+    private MetricRegistry metricRegistry;
+
+    public static void main(String[] args) throws Exception {
+        new FrontendEnvironment().setupServers();
+    }
+
+    @Setup(Level.Trial)
+    public void setupServers() throws Exception {
+        zookeeperStarter = new ZookeeperStarter(ZOOKEEPER_PORT, "localhost");
+        zookeeperStarter.start();
+
+        kafkaStarter = new KafkaStarter("/kafka.properties");
+        kafkaStarter.start();
+
+        hermesFrontend = HermesFrontend.frontend().build();
+        hermesFrontend.start();
+
+        GroupRepository groupRepository = hermesFrontend.getService(GroupRepository.class);
+        TopicRepository topicRepository = hermesFrontend.getService(TopicRepository.class);
+        groupRepository.createGroup(Group.from("bench"));
+        topicRepository.createTopic(topic("bench.topic")
+                .withContentType(AVRO)
+                .withMessageSchema(loadMessageResource("schema"))
+                .build());
+    }
+
+    @Setup(Level.Trial)
+    public void setupPublisher() throws Exception {
+        metricRegistry = new MetricRegistry();
+
+        String messageBody = loadMessageResource("completeMessage");
+        publisher = new HermesPublisher(MAX_CONNECTIONS_PER_ROUTE, "http://localhost:8080/topics/bench.topic", messageBody, metricRegistry);
+    }
+
+    @TearDown(Level.Trial)
+    public void shutdownServers() throws Exception {
+        hermesFrontend.stop();
+        kafkaStarter.stop();
+        zookeeperStarter.stop();
+    }
+
+    @TearDown(Level.Trial)
+    public void shutdownPublisherAndReportMetrics() throws Exception {
+        reportMetrics();
+
+        shutdownPublisher();
+    }
+
+    public void shutdownKafka() throws Exception {
+        kafkaStarter.stop();
+    }
+
+    public void shutdownPublisher() throws IOException {
+        publisher.stop();
+    }
+
+    public HermesPublisher publisher() {
+        return publisher;
+    }
+
+    public static String loadMessageResource(String name) throws IOException {
+        return IOUtils.toString(FrontendEnvironment.class.getResourceAsStream(String.format("/message/%s.json", name)));
+    }
+
+    private void reportMetrics() {
+        System.out.println("\n");
+        metricRegistry.getCounters().entrySet().forEach(e -> System.out.printf(e.getKey() + ": " + e.getValue().getCount() + "\n"));
```

Comment:
If we use logger then we could append benchmark results to a file for instance...